### PR TITLE
Pin redir_protocols

### PR DESCRIPTION
**Pull request by yoannchaudet**
This issue addresses https://github.com/github/pages-engineering/issues/5244.

Limit the redirection protocols we follow to HTTP/HTTPS only.

I did not find a way to test that with the mocking lib in the way. The [`redirect_count`](https://www.rubydoc.info/github/typhoeus/typhoeus/Typhoeus/Response/Informations#redirect_count-instance_method) method also does not exactly returned what we needed. When you cannot fake it, just do it...

So to test the proper behaviors, we start a few real TCP servers in parallel to running the tests. We can confirm we get the behavior we need based on a log file written by the TCP servers. It is a bit clunky but does validate the behavior we are getting is the one we expect.

### CODE REVIEW

**Review by parkr — APPROVED** [review]
A test would be ideal if there's behavior you're trying to avoid and want to prevent regressions but LGTM 😀

**Comment by yoannchaudet** [comment]
That one will be annoying to test but I'll try to come up with something. I am getting the changes themselves vetted first ;)

File: spec/github_pages_health_check/domain_spec.rb [hunk 1]
```diff
@@ -663,6 +663,35 @@
     end
   end
 
+  context "no protocol switch" do
+    let(:log_file) { "/tmp/bad-redirection.log" }
+    before do
+      File.open(log_file, "w")
```
Comment (yoannchaudet) [inline spec/github_pages_health_check/domain_spec.rb:669]:
```suggestion
      File.open(log_file, "w") # truncate log file after each test
```

File: script/cibuild-docker [hunk 2]
```diff
@@ -4,8 +4,8 @@
 docker build -t github-pages-health-check --build-arg RUBY_VERSION=$RUBY_VERSION .
 if [ -n "$DEBUG" ]; then
   # Run a shell.
-  docker run -it --rm -v $(pwd):/app/github-pages-health-check github-pages-health-check
+  docker run -e RUBY_VERSION -it --rm -v $(pwd):/app/github-pages-health-check github-pages-health-check
```
Comment (yoannchaudet) [inline script/cibuild-docker:7]:
```suggestion
  docker run -it --rm -v $(pwd):/app/github-pages-health-check github-pages-health-check
```

File: script/cibuild-docker [hunk 3]
```diff
@@ -4,8 +4,8 @@
...
 else
   # Run CI
-  docker run --rm github-pages-health-check script/cibuild --profile --fail-fast
+  docker run -e RUBY_VERSION --rm github-pages-health-check script/cibuild --profile --fail-fast
```
Comment (yoannchaudet) [inline script/cibuild-docker:10]:
```suggestion
  docker run --rm github-pages-health-check script/cibuild --profile --fail-fast
```

File: script/cibuild [hunk 4]
```diff
@@ -4,6 +4,8 @@ set -ex
 
 script/bootstrap
 
+echo "Using RUBY_VERSION=$RUBY_VERSION"
+
```
Comment (yoannchaudet) [inline script/cibuild:8]:
```suggestion
```

**Review by tsusdere — APPROVED** [review]

**Pull request merged**